sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3752,11 +3752,38 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
   PI_ASSERT((WorkDim > 0) && (WorkDim < 4), PI_INVALID_WORK_DIMENSION);
 
   if (GlobalWorkOffset != NULL) {
-    for (pi_uint32 i = 0; i < WorkDim; i++) {
-      if (GlobalWorkOffset[i] != 0) {
-        return PI_INVALID_VALUE;
+    uint32_t Count = 0;
+    ZE_CALL(zeDriverGetExtensionProperties,
+            (Queue->Device->Platform->ZeDriver, &Count, nullptr));
+
+    if (Count == 0) {
+      zePrint("No extensions supported on this driver\n");
+      return PI_INVALID_VALUE;
+    }
+
+    std::vector<ze_driver_extension_properties_t> Extensions(Count);
+    ZE_CALL(zeDriverGetExtensionProperties,
+            (Queue->Device->Platform->ZeDriver, &Count, Extensions.data()));
+
+    bool ExtensionFound = false;
+    for (uint32_t i = 0; i < Extensions.size(); i++) {
+      if (strncmp(Extensions[i].name, ZE_GLOBAL_OFFSET_EXP_NAME,
+                  strlen(ZE_GLOBAL_OFFSET_EXP_NAME)) == 0) {
+        if (Extensions[i].version == ZE_GLOBAL_OFFSET_EXP_VERSION_1_0) {
+          ExtensionFound = true;
+          break;
+        }
       }
     }
+
+    if (ExtensionFound == false) {
+      zePrint("No global offset extension found on this driver\n");
+      return PI_INVALID_VALUE;
+    }
+
+    ZE_CALL(zeKernelSetGlobalOffsetExp,
```

Comment:
So this will not build with the L0 loader that doesn't have the `zeKernelSetGlobalOffsetExp`, right? What is the reason to really search for extension support above? Can't we rely on this call to fail (return error) if the driver doesn't have the proper support?

Reply:
I also wanted to have a discussion on it. On the experimental support patch, Gail added the search and I assumed that there might be some dependency with the version and name. If that's not the case, I think that can be removed. 
@jandres742 could you please review and confirm that the search is unnecessary?

Reply:
so, in order to build you **must** have the L0 1.1 headers. Otherwise as @smaslov-intel  says, building would fail, as the compiler wont find a definition of `zeKernelSetGlobalOffsetExp`. However, this is an experimental extension, so even if it is defined in the headers, it might not be supported by the driver.

So the search is to confirm at runtime the target driver has support for this extension, but you need to build this with L0 loader 1.1. otherwise it would fail.

Reply:
> this is an experimental extension, so even if it is defined in the headers, it might not be supported by the driver.

@jandres742: So, wouldn't driver return the "unsupported" error, which we'd use to know that this extension is not supported (instead of doing the search)?

Reply:
@smaslov-intel  that's if we have at least the C++-stub implemented. But if the driver doesnt even have that, then you wouldn't be able to even get that error code.

Reply:
@jandres742 how would the fail look like in this case? Where would the Level-Zero loader route the call to `zeKernelSetGlobalOffsetExp`?

Reply:
Right now it would return UNSUPPORTED_VERSION. My suggestion is to better still perform the search, as that is the recommended way from the spec to use extensions. That will help you in the future extend that mechanism when you need to query for instance which version of the extension you want and which one is provided.

Reply:
OK, so we **could** rely on the error reported, but would like to be more robust. I am fine with querying supported extensions, but that should certainly be cached.

Reply:
So, if I understood this correctly, we should keep the search? @jandres742

Reply:
Correct.